src/error.rs

```diff
@@ -0,0 +1,107 @@
+use std::error::Error;
+use std::fmt;
+use std::io;
+use color::ColorType;
+use image::ImageFormat;
+
+/// An enumeration of Image errors
+#[derive(Debug)]
+pub enum ImageError {
+    /// The Image is not formatted properly
+    FormatError(String),
```

Comment:
Perhaps `FileCorrupt(String)`?

Reply:
Yeah, that would probably be a bit more specific. Only problem is that I use this variant for encoders as well, but with some sort of `InvalidData` it could be a nice combination.

Reply:
One more thing we could be clearer on here: If the format contains a `length` field but the file ends prematurely. Is it

* `FormatError`
* `NotEnoughData`
* `io::ErrorKind::UnexpectedEof`, the return when reading that data with `r.read_exact(&mut slice)?`

Reply:
This doesn't need a definitive answer, but definitely examples in the doc to make choosing a variant easier in all implementations.